internal/karma/karma.conf.js

```diff
@@ -4,6 +4,86 @@ const path = require('path');
 const fs = require('fs');
 const tmp = require('tmp');
 
+const WORKSPACE_NAME = 'TMPL_workspace_name';
+
+// On Windows, runfiles will not be in the runfiles folder but instead
+// there is a MANIFEST file which maps the runfiles for the test
+// to their location on disk. Bazel provides a RUNFILE_MANIFEST_FILE environment
+// variable which is set the location of the MANIFEST file during test execution.
+// If a MANIFEST file is found, we remap the test files to their
+// location on disk using the MANIFEST file.
+let manifest = null;
+if (process.env.RUNFILES_MANIFEST_ONLY === '1' &&
+  process.env.RUNFILES_MANIFEST_FILE && fs.existsSync(process.env.RUNFILES_MANIFEST_FILE)) {
+  // MANIFEST file contains the one runfile mapping per line seperated
+  // a space. For example:
+  // rxjs/operators.js /private/var/tmp/.../external/rxjs/operators.js
+  // The file is parsed here into a map of for easy lookup
+  manifest = {};
+  for (l of fs.readFileSync(process.env.RUNFILES_MANIFEST_FILE, 'utf8').split('\n')) {
+    const m = l.split(' ');
+    manifest[m[0]] = m[1];
+  }
+}
+
+function isDirectory(res) {
+  try {
+    return fs.statSync(res).isDirectory();
+  } catch (e) {
+    return false;
+  }
+}
+
+function readDir(dir) {
+  return fs.statSync(dir).isDirectory() ?
+    Array.prototype.concat(...fs.readdirSync(dir).map(f => readDir(path.join(dir, f)))) :
+    dir.replace(/\\/g, '/');
+}
+
+function resolveManifestFile(runfilesManifest, res) {
+  const maybe = runfilesManifest[res];
+  if (maybe) {
```

Comment:
can we use the runfiles JS library support rather than copy-paste code? @gregmagolan  can help

Reply:
I didn't know we had one, and was going to suggest making it. Yeah I think we should, and we should use it throughout everywhere. It needs to be able to look up runfiles generally and not just node resolution.

Reply:
actually it seems like we should land https://github.com/bazelbuild/rules_typescript/pull/278 first